app/models/document.rb

```diff
@@ -133,9 +91,7 @@ def documents_count
     errors.add(:document, "Total documents exceed maximum of #{max_doc_count}. This document has not been uploaded.")
   end
 
-  def transform_cryptic_paperclip_error
-    return unless errors[:document].include?('has contents that are not what they are reported to be')
-    errors[:document].delete('has contents that are not what they are reported to be')
-    errors[:document] << 'The contents of the file do not match the file extension'
+  def create_preview_document
+    DocumentConverterService.new(document).to(converted_preview_document)
```

Comment:
could this just be
```
DocumentConverterService.new(document, converted_preview_document).call
```
see comment below and [this blog example](https://www.toptal.com/ruby-on-rails/rails-service-objects-tutorial)